src/app/youtube/components/search/search-card/search-card.component.ts

```diff
@@ -1,12 +1,31 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
+import { CONSTANTS } from 'src/app/shared/models/constants';
+import { ICustomCard } from 'src/app/youtube/models/custom-card.model';
 import { ISearchCard } from 'src/app/youtube/models/search-card.model';
 
 @Component({
   selector: 'app-search-card',
   templateUrl: './search-card.component.html',
   styleUrls: ['./search-card.component.css'],
 })
-export class SearchCardComponent {
-  @Input() public item: ISearchCard = Object.assign({});
+export class SearchCardComponent implements OnInit {
+  @Input() public item: ISearchCard | ICustomCard = Object.assign({});
   @Input() public search = '';
+  public itemData: ICustomCard = {} as ICustomCard;
+
+  public ngOnInit() {
+    if (this.isISearchCard(this.item)) {
+      this.itemData.title = this.item.snippet.title;
+      this.itemData.date = this.item.snippet.publishedAt;
+      this.itemData.image = this.item.snippet.thumbnails.medium.url;
+      this.itemData.id = typeof this.item.id === 'string' ? this.item.id : this.item.id.videoId;
+    } else {
```

Comment:
I'd use this construction:
this.itemData={...this.item,
title: this.item.snippet.title, ..}